core/src/main/java/hudson/model/AsyncPeriodicWork.java

```diff
@@ -87,34 +88,32 @@ public final void doRun() {
                 logger.log(this.getSlowLoggingLevel(), "{0} thread is still running. Execution aborted.", name);
                 return;
             }
-            thread = new Thread(new Runnable() {
-                public void run() {
-                    logger.log(getNormalLoggingLevel(), "Started {0}", name);
-                    long startTime = System.currentTimeMillis();
-                    long stopTime;
-
-                    StreamTaskListener l = createListener();
-                    try {
-                        l.getLogger().printf("Started at %tc%n", new Date(startTime));
-                        ACL.impersonate(ACL.SYSTEM);
+            thread = new Thread(() -> {
+                logger.log(getNormalLoggingLevel(), "Started {0}", name);
+                long startTime = System.currentTimeMillis();
+                long stopTime;
 
+                StreamTaskListener l = createListener();
+                try {
+                    l.getLogger().printf("Started at %tc%n", new Date(startTime));
+                    try (ACLContext ctx = ACL.as(ACL.SYSTEM)) {
```

Comment:
can't we move this try up to the one above?

Reply:
I'm pretty sure i fixed this I'm not sure why the github diff says otherwise

Reply:
Seems to be fixed here =/
https://github.com/jenkinsci/jenkins/blob/7bdc9308fa0624211452c3bbb1cbdb840c9978d2/core/src/main/java/hudson/model/AsyncAperiodicWork.java

Reply:
Copy-pasta 🍝 Let's wait till @jsoref 's tools can cover these cases :D